execution/inc/leatherman/execution/execution.hpp

```diff
@@ -187,6 +187,16 @@ namespace leatherman { namespace execution {
     };
 
     /**
+     * Encapsulates return value from executing a process: whether or not the command succeeded, output from stdout, output from stderr (if not redirected), and the exit code.
+     */
+    struct exec_result
```

Comment:
I'd prefer this called `execution_result` or simply `result` since we're in the `execution` namespace, but can live with `exec_result`.